src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,149 @@
+};
+
+class IOUModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.navigateToPreviousStep = this.navigateToPreviousStep.bind(this);
+        this.navigateToNextStep = this.navigateToNextStep.bind(this);
+
+        this.state = {
+            steps: [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm],
+            currentStepIndex: 0,
+            currentStepTitle: Steps.IOUAmount,
+            hasMultipleParticipants: this.props.currentURL === ROUTES.IOU_BILL_SPLIT,
+        };
+    }
+
+    componentDidMount() {
+        getPreferredCurrency();
+    }
+
+    /**
+     * Navigate to the next IOU step if possible
+     */
+    navigateToPreviousStep() {
+        if (this.state.currentStepIndex <= 0) {
+            return;
+        }
+        this.setState(prevState => ({
+            currentStepIndex: prevState.currentStepIndex - 1,
+            currentStepTitle: prevState.steps[prevState.currentStepIndex - 1] || '',
+        }));
+    }
+
+    /**
+     * Navigate to the previous IOU step if possible
+     */
+    navigateToNextStep() {
+        if (this.state.currentStepIndex >= this.state.steps.length - 1) {
+            return;
+        }
+        this.setState(prevState => ({
+            currentStepIndex: prevState.currentStepIndex + 1,
+            currentStepTitle: prevState.steps[prevState.currentStepIndex + 1] || '',
+        }));
+    }
+
+    render() {
+        return (
+            <>
+                <View style={[styles.headerBar, true && styles.borderBottom]}>
+                    <View style={[
+                        styles.dFlex,
+                        styles.flexRow,
+                        styles.alignItemsCenter,
+                        styles.flexGrow1,
+                        styles.justifyContentBetween,
+                        styles.overflowHidden,
+                    ]}
+                    >
+                        {this.state.currentStepIndex > 0
+                        && (
+                            <TouchableOpacity
+                                onPress={this.navigateToPreviousStep}
+                                style={[styles.touchableButtonImage]}
+                            >
+                                <Icon src={BackArrow} />
+                            </TouchableOpacity>
+                        )}
+                        <Header title={this.state.currentStepTitle} />
+                        <View style={[styles.reportOptions, styles.flexRow]}>
+                            <TouchableOpacity
+                                onPress={redirectToLastReport}
+                                style={[styles.touchableButtonImage]}
+                            >
+                                <Icon src={Close} />
+                            </TouchableOpacity>
+                        </View>
+                    </View>
+                </View>
+                {this.state.steps[this.state.currentStepIndex] === Steps.IOUAmount
```

Comment:
OK, I seem to have lost the ability to comment on the previous thread about this, so I'm copying over your last comment:

> Totally agree with your concerns and the mentioned benefits.

> This Component was actually planned with react-navigation in mind from the beginning, once available we'll make each page a Tab.Screen component and will take advantage of the above mentioned benefits. An additional benifit is that we'll get native page-to-page animations for free too! (using tab-navigation).

So, does that mean that you don't want to use `<Route>` here, and this will be done later? Or that you should switch to using `<Route>` here which will make the transition to react-navigation easier? I wasn't quite sure what the conclusion of your comment was.

Reply:
> So, does that mean that you don't want to use <Route> here, and this will be done later? Or that you should switch to using <Route> here which will make the transition to react-navigation easier? I wasn't quite sure what the conclusion of your comment was.

My bad, let me clarify. I suggest that this will remain as in until the react-navigation PR is merged. As this issue is currently the main blocking issue for IOU, I think it's best to move forward rather than implementing the temporary route -- which might end up being replaced with the minimal effort `react-native-navigation` changes prior to the release of the project.

However, in the case where the IOU project is ready for release and `react-native-navigation` has not yet been merged, it might be a good idea to implement route temporarily. I have created an issue for this task ([Improve IOU step navigation](https://github.com/Expensify/Expensify/issues/156712)) and attached it to the [[IOU] project master tracking issue](https://github.com/Expensify/Expensify/issues/148974).

Reply:
OK, got it. Thanks! That plan sounds fine to me. Thank you for creating a clean up issue.

Reply:
I maybe missed some of this convo and the outcome seems fine. But I would have leaned towards using routes for consistency. Especially if we are planning on deep linking to a particular step in in this flow in the future. In `react-navigation` land every screen is also a route so it's unclear to me why we are building a stepper instead of using a route based navigation.

Reply:
>  In react-navigation land every screen is also a route

Ah, this was the piece I was missing. The Design Doc plan was simply to wrap the steps with  `<Tab.Navigator>` & `<Tab.Screen>` components. But this doesn't cover web routing...

> Especially if we are planning on deep linking to a particular step in in this flow in the future.

The point remains valid, but I think we can rule this out for certain. Users should not be able to navigate forward through the IOU flow in any way, as this skips our validation  -- which I think would become possible with web routes? 

Anyway, I'll update the [improve navigation](https://github.com/Expensify/Expensify/issues/156712) issue with this info to ensure we include route support as part of the navigation improvements.

Reply:
>Users should not be able to navigate forward through the IOU flow in any way, as this skips our validation -- which I think would become possible with web routes

To clarify, are you saying that we do not want to use routes here since the user should not be able to navigate between screens? I'm not 100% sure how a tab navigator is handled for `react-navigation` on web but it sounds like we don't need to worry about this right now.

Reply:
Not quite, I just wanted to mention that deep linking into the flow isn't something we need to implement yet (as that seemed to be a main argument for using Routes for step navigation). 

Are you happy for us to move the discussion surrounding routes/react-navigation to the [linked issue](https://github.com/Expensify/Expensify/issues/156712), to separate it from this PR? These are important problems to solve for the IOU project, and I believe we're aligned on the need for `react-native-navigation` with Routes. But I don't think it needs to be solved on this PR.

The first step will be for me to take another look at the react-navigation PR to understand what is required for making the IOUModal compatible. Your suggestions/feedback on this would be much appreciated.

Reply:
That sounds great to me. It's unclear whether we would need/want to do this any other way than you are handling things now.